src/Model/PropertyGo.cs

```diff
@@ -11,7 +11,20 @@ public class PropertyGo : Property
     {
         public PropertyGo()
         {
-
+            Name.OnGet += v =>

```

Comment:
How does this fix anything, is it due to some ordering issue during transformation?

Reply:
@jhendrixMSFT short answer yes. We are doing some transformation on the composite type transformer where we set the property name to be the type name for flattened types ( which will be embedded types for us ).  When generating the whole sdk hat produced issues because later we were changing the type name. 

I removed that part here and move this check here.